packages/kit/src/vite/utils.js

```diff
@@ -4,13 +4,15 @@ import { pathToFileURL } from 'url';
 import { get_runtime_path } from '../core/utils.js';
 
 /**
+ * @param {import('vite').ConfigEnv} config_env
  * @return {Promise<import('vite').UserConfig>}
  */
-export async function get_vite_config() {
+export async function get_vite_config(config_env) {
 	for (const file of ['vite.config.js', 'vite.config.mjs', 'vite.config.cjs']) {
 		if (fs.existsSync(file)) {
-			const config = await import(pathToFileURL(file).toString());
-			return config.default || config;
+			const m = await import(pathToFileURL(file).toString());
+			const config = m.default || m;
+			return typeof config === 'function' ? config(config_env) : config;
 		}
 	}
 	throw new Error('Could not find vite.config.js');
```

Comment:
Maybe we could use `resolveConfig` from `vite` to search for the vite config? It handles typescript too so that could make this part more aligned to Vite.

Though we could also do this separately.

Reply:
Would I call it like this?

```js
export async function get_vite_config(config_env) {
	// @ts-ignore
	return await resolveConfig({}, config_env.command, config_env.mode);
}
```

That was my best guess. But `resolveConfig` returns a `ResolvedConfig` rather than a `UserConfig`. If I tell TypeScript not to worry about that, I get a runtime error `[vite-plugin-svelte-kit] Cannot read properties of null (reading 'constructor')` so apparently one of the things in `UserConfig` but not `ResolvedConfig` is needed here.

Reply:
Yeah I think that would work. A SvelteKit project now would always need a Vite config so the error should never happen.

Reply:
Updated in ccee0571df341b9f9c6dda81cba54f119812f21b.